spec/models/upload_spec.rb

```diff
@@ -78,6 +78,17 @@
     expect(upload.thumbnail_width).to eq(nil)
   end
 
+  it 'returns error when image resolution is to big' do
+    begin
+      SiteSetting.max_image_megapixels = 10
+      upload = UploadCreator.new(huge_image, "image.png").create_for(user_id)
+      expect(upload.id).to be_nil
+      expect(upload.errors.messages[:base].first).to eq("Sorry, the image you are trying to upload is too large (maximum dimension is 20-megapixels), please resize it and try again.")
```

Comment:
I usually prefer to just use the translation so that changing the translation doesn't break the spec.